core/src/test/java/com/expedia/adaptivealerting/core/detector/PewmaOutlierDetectorTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2018 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.core.detector;
+
+import com.expedia.adaptivealerting.core.OutlierLevel;
+import com.opencsv.CSVReader;
+import com.opencsv.bean.CsvToBeanBuilder;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+
+import static com.expedia.adaptivealerting.core.util.MathUtil.isApproximatelyEqual;
+import static java.lang.Math.sqrt;
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+
+public class PewmaOutlierDetectorTest {
+    private static final double WEAK_THRESHOLD = 2.0;
+    private static final double STRONG_THRESHOLD = 3.0;
+    private static final int DEFAULT_TRAINING_LENGTH = 30;
```

Comment:
Nice. We should probably add this training period to the EWMA and PEWMA evaluators themselves. (Not a blocker for the PR; just an idea for future enhancement.)